src/ol_osd_module.c

```diff
@@ -698,6 +698,13 @@ ol_osd_module_set_lrc (struct OlDisplayModule *module, OlLrc *lrc_file)
   }
 
   priv->lrc = lrc_file;
+
+  if (lrc_file != NULL)
+  {
+    /* Until we are called back with the current play time, display the first
```

Comment:
In theory, yes, if set_lrc() is called when changing mode (as you point out below), it will cause the display to jump back to 0. In practice though it doesn't matter, because the display is refreshed both in the playing and paused states, and therefore it will show the right position a fraction of second later. If stopped, the display is not refreshed, but in this case the time is 0 so that's fine.

That said, this is clearly not very clean. I like your suggestion, this sounds rather easy to implement and much cleaner. I'll try and implement it in this PR.

Speaking of this, is there a strong reason why the position is updated even while the player is paused (see `_status_changed_cb()`)? It seems to me that if I separate setting time and updating the display, this shouldn't be needed anymore.